python/graphstorm/gconstruct/file_io.py

```diff
@@ -88,6 +88,8 @@ def expand_wildcard(data_files: List[str]) -> List[str]:
 
     """
     expanded_files = []
+    if len(data_files) == 1 and os.path.isdir(data_files[0]):
+        data_files = [os.path.join(data_files[0], "*")]
```

Comment:
This adds support for path prefixes to gconstruct (`path/to/files/`), allowing the exact same config to be used for both gconstruct and gsporcessing

Reply:
I would suggest we create a standalone PR for this fix and add some unit tests.